test/jasmine/tests/plotschema_test.js

```diff
@@ -112,17 +113,31 @@ describe('plot schema', function() {
         expect(list).toEqual(astrs);
     });
 
-    it('layout.annotations and layout.shapes should contain `items`', function() {
-        var astrs = ['annotations', 'shapes'];
+    it('should convert _isLinkedToArray attributes to items object', function() {
+        var astrs = [
+            'annotations', 'shapes',
+            'xaxis.rangeselector.buttons', 'yaxis.rangeselector.buttons'
+        ];
 
         astrs.forEach(function(astr) {
-            expect(
-                isPlainObject(
-                    Lib.nestedProperty(
-                        plotSchema.layout.layoutAttributes, astr
-                    ).get().items
-                )
-            ).toBe(true);
+            var np = Lib.nestedProperty(
+                plotSchema.layout.layoutAttributes, astr
+            );
+
+            var name = np.parts[np.parts.length - 1],
+                itemName = name.substr(0, name.length - 1);
+
+            var itemsObj = np.get().items,
+                itemObj = itemsObj[itemName];
+
+            expect(isPlainObject(itemsObj)).toBe(true);
+            expect(itemsObj.role).toBeUndefined();
+            expect(Object.keys(itemsObj).length).toEqual(1);
+            expect(isPlainObject(itemObj)).toBe(true);
+            expect(itemObj.role).toBe('object');
+
+            var role = np.get().role;
+            expect(role).toEqual('object');
```

Comment:
@delekru <https://github.com/delekru> @etpinard @chriddyp
Can we please use Highchart's button styling? I think they did a good job
and its what users are familiar with:
http://www.highcharts.com/stock/demo/basic-line

On Fri, Apr 1, 2016 at 4:20 PM, Étienne Tétreault-Pinard <
notifications@github.com> wrote:

> In test/jasmine/tests/plotschema_test.js
> <https://github.com/plotly/plotly.js/pull/373#discussion_r58269266>:
>
> > +            );
> > +
> > +            var name = np.parts[np.parts.length - 1],
> > +                itemName = name.substr(0, name.length - 1);
> > +
> > +            var itemsObj = np.get().items,
> > +                itemObj = itemsObj[itemName];
> > +
> > +            expect(isPlainObject(itemsObj)).toBe(true);
> > +            expect(itemsObj.role).toBeUndefined();
> > +            expect(Object.keys(itemsObj).length).toEqual(1);
> > +            expect(isPlainObject(itemObj)).toBe(true);
> > +            expect(itemObj.role).toBe('object');
> > +
> > +            var role = np.get().role;
> > +            expect(role).toEqual('object');
>
> @cldougl <https://github.com/cldougl> @theengineear
> <https://github.com/theengineear> [image: :arrow_double_up:] needs to be
> satisfied for plotly.py to build to correct graph object.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/plotly/plotly.js/pull/373/files/a671683931ac3856674bce3af7b5d988c0857cb5..1042f4f656c3e1ac14a365477a93c6d66d44b855#r58269266>
>